Kristiyan96/bulkit
# Delete images (#34)

* Develop (#33)

* Fix deploy (#22)

* Clean up

* Small fixes

* Clean up

* Formatting and style

* Test issues

* Update test

* Test next

* Auth test

* Test

* Test

* Test

* Test

* Simplify footer

* Small UI improvements

* Improve footer

* Clean up

* Test if works

* Tests styles

* Test ui

* Test deploy

* Test

* Test

* Test

* Revert changes

* Test

* Test

* Test

* Test

* Test

* Test

* Test

* Test

* Test

* Test

* Test

* Test

* Test

* Test

* Test

* Fix height of buttons

* Fix deploy (#24)

* Test

* Test

* Test

* Test

* Test

* Fix deploy inconsistencies (#26)

* Test

* Test

* Add menu ids

* Use BASE_URL and add to documentation (#28)

* Add permissions (#29)

* Remove pagination

* Send profile in Bearer token

* Create permissions for collections and links

* Create permissions for profiles

* Add role to users

* Verify users are logged in for certain routes

* Improve routes set up and add comments

* Add canUpdate permissions

* Set up permissions for collections

* Fix link permissions

* Implement permissions for links

* Fix profile data issue

* Update collections controller

* Update collection routes

* Improve links controller

* Fix issue

* Create collections and links middleware

* Use middlewares in routes

* Move basicAuth to auth middleware

* Fix exports/imports

* Set profile in jwt token

* Fix small issues

* Fix small issues

* Improve links controller

* Improve middleware

* Create a profile middleware

* Improve permissions

* Improve profile controller

* Improve permissions

* Fix profile controller

* Profile middleware and routes

* Format code

* Fix "Cannot set headers after they are sent"

* Fix permission issue

* Bug fixes

* Fix bugs

* Fix update issues

* Fix bug

* Fix null value issue

* Fix bugs

* Login and signup (#31)

* Create a signup router and component

* Point to signin and signup

* Redirect to login after logout

* Customization panel (#32)

* Add background and format

* Add customize button in header

* Install pexels api

* Create customization drawer

* Create an image model

* Create Image model

* Create Image routes

* Create images permissions

* Create images middleware

* Create images controller

* Update Image model

* Add image constants

* Create images actions

* Create AddImage component

* Import image routes

* Add image routes  on client

* Make add icon bigger

* Add wallpaper dialog

* Display AddButton in image gallery

* Fix bug

* Create images reducer

* Import images reducer

* Display user images in drawer

* Install mui/lab

* Improve masonry

* Create UI reducer

* Set body background programatically

* Ui improvements

* Add nginx config

* Add photographer details to wallpaper form

* Add credits fields in image form

* Create customize app bar

* Add empty icon

* UI improvements

* Mobile UI

* Mobile UI

* Improve wording

* Create 401 interceptor

* Small improvements

* Export history from App

* Use forbidden error code

* Improve image ui

* Simplify

* Simplify

* Delete an image

* Improve UI

* Allow user to edit images

* Image preview component

* Ui improvement

* Fix bug

* Prevent selecting the current image
